x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_logic.ts

```diff
@@ -308,35 +297,42 @@ export const ResultSettingsLogic = kea<MakeLogicType<ResultSettingsValues, Resul
         flashAPIErrors(e);
       }
     },
-    saveResultSettings: async ({ resultFields }) => {
-      actions.saving();
+    confirmResetAllFields: () => {
+      if (window.confirm(SAVE_CONFIRMATION_MESSAGE)) {
+        actions.resetAllFields();
+      }
+    },
+    saveResultSettings: async () => {
+      if (window.confirm(SAVE_CONFIRMATION_MESSAGE)) {
+        actions.saving();
 
-      const { http } = HttpLogic.values;
-      const { engineName } = EngineLogic.values;
-      const url = `/api/app_search/engines/${engineName}/result_settings`;
+        const { http } = HttpLogic.values;
+        const { engineName } = EngineLogic.values;
+        const url = `/api/app_search/engines/${engineName}/result_settings`;
 
-      actions.saving();
+        actions.saving();
 
-      let response;
-      try {
-        response = await http.put(url, {
-          body: JSON.stringify({
-            result_fields: resultFields,
-          }),
-        });
-      } catch (e) {
-        flashAPIErrors(e);
-      }
+        let response;
+        try {
+          response = await http.put(url, {
+            body: JSON.stringify({
+              result_fields: values.reducedServerResultFields,
```

Comment:
This previous required `reducedServerResultFields` to be passed in from the consuming component, which seemed unnecessary, so I changed it to grab the value directly from state.